src/libprojectM/Renderer/StaticGlShaders.cpp

```diff
@@ -770,16 +772,74 @@ StaticGlShaders::StaticGlShaders(bool use_gles) : use_gles_(use_gles) {
 }
 
 StaticGlShaders::GlslVersion StaticGlShaders::QueryGlslVersion() {
+    /* In the linux desktop environment with --use-gles configured, the parsing of the GL_SHADING_LANGUAGE_VERSION
+    string comes back as "OpenGL ES GLSL ES 3"
+    And I think this was supposed to be parsing somethign like
+    "3.10 etc etc"
+    This caused exceptions to be raised in the std::stoi section;
+    
+    So - The parsing will look for <anything> <number> ['.' <number>] [<anything else>]
+    and will default to 3.0 for the version in case of errors    
+    */
+    
+    int major = 3;  /* 3.0 is default */
+    int minor = 0;
+
     std::string glsl_version_string = reinterpret_cast<const char *>(
         glGetString(GL_SHADING_LANGUAGE_VERSION));
-    size_t dot_location = glsl_version_string.find('.');
-    size_t end_location = glsl_version_string.find(' ');
-    auto major_string = glsl_version_string.substr(0, dot_location);
-    auto minor_string = glsl_version_string.substr(dot_location + 1,
-                                                   end_location - dot_location);
-    int major = std::stoi(major_string);
-    int minor = std::stoi(minor_string);
 
+    size_t version_len = glsl_version_string.length();
+    /* make a c version of the string and do the conversion to integers manually just for this case */
+    if (version_len) { // find the number
+      size_t position = 0; 
+      char *cstr = new char [version_len+1];
+
+      strcpy(cstr,glsl_version_string.c_str()); 
+      
+      /* scan the anything before the number */
+      while (position<version_len) {
+        char ch = cstr[position];
+	if ((ch >= '0') || (ch <= '9')) {
```

Comment:
I think this is always true?

Reply:
OOPS.
The || should be && here and two other places.